### PR TITLE
Rebalances Metalgen to prevent issues with material export datums.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

#49991 had a bit of an unintended consequence in that I had never heard of a little thing called metalgen. Metalgen could be used with this new system to completely invalidate the existing export value balance in place with datum materials.
So, primary changes:
* Metalgen cannot be used on stack items, in order for you to convert very very low value materials (Like plastic) into very high value items (Like plasma), netting something like a metric fuckton of profit for less than a full unit of metalgen.
* Objects who have had their materials changed with metalgen are no longer valid for export.

## Why It's Good For The Game

Otherwise, a very very valid way to rack up thousands of credits in a matter of minutes exists, and I've already managed to prove oranges right once this morning so I'm willing to disable this for now until I see the actual effects on material exports allowing most everything to be exportable.

## Changelog
:cl:
balance: Metalgen cannot be used to convert stacks of materials into different materials.
balance: Objects sprayed in metalgen cannot be exported for their material profit, because Nanotrasen doesn't believe in alchemy.
/:cl:

### CODE REVIEW

**Comment by Eskjjlj** [comment]
Being able to use metalgen is a miracle considering how rare wittel is. This nerf is unnecessary.
Wittel has 1/27 chance to appear in a geyser on lavaland that you need to unplug with a reinforced plunger and then build a liquid pump over it with a plumbing constructor. So it takes a lot of time and effort to find look for it and it's not even guaranteed you will find wittel before the round ends.
Once you have wittel you need to turn it into metalgen with unstable mutagen and bluespace dust and imprint it with 40u of a material in liquid form using stabilized dark matter.

Personally after spending a dozen hours looking for it I have only found wittel twice and did not have enough time to use it since the rounds were about to end.

So this shows how complex the process of getting metalgen is and why the rewards should be so great. 
Having someone find wittel and make cargo rich is something that can only happen once every 50 rounds if a dedicated player spends time looking for the wittel geyser round after round.

**Comment by Tlaltecuhtli** [comment]
isnt this already fixed by flexibility(read as 1 stack gold is worth 7k next one is 1k then 100 credits and so on)  for everything except plasma? make the wittel turn plasma into a plasma subtype which is the same except it has flexibiliy so it doesnt break the game too much

**Comment by ATH1909** [comment]
I think what Eskjjlj is trying to say is that a super-rare (and effort-intensive to obtain once you've found it) miracle chemical that can transmute iron into gold (or plasma or whatever) SHOULD make you shitloads of cash.

File: code/modules/cargo/exports.dm [hunk 1]
```diff
@@ -145,7 +145,8 @@ Credit dupes that require a lot of manual work shouldn't be removed, unless they
 	var/amount = get_amount(O)
 	///Utilized in the pricetag component. Splits the object's profit when it has a pricetag by the specified amount.
 	var/profit_ratio = 0
-
+	if(O.material_flags & MATERIAL_NO_EXPORT)//Anything and everything to prevent literal alchemy.
```
Comment (Qustinnus) [inline code/modules/cargo/exports.dm:148]: This needs more feedback els noone will know why this happens

**Comment by Spepbuthewranglesgithub** [comment]
still worse than selling gasses

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 2]
```diff
@@ -2081,7 +2081,8 @@
 	if(!metal_ref)
 		return
 	var/metal_amount = 0
-
+	if(istype(A,/obj/item/stack/sheet))
```
Comment (Qustinnus) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:2084]: I don't see the need of doing this if you are already making it so you can't sell it.

**Comment by ArcaneMusic** [comment]
After having a few days to think on this one, this was a bit of a kneejerk reaction considering how exceptionally rare this is in the first place. If anything, this is used so little that it's better off to leave it for the time being.

**Pull request closed without merging**